.github/workflows/update-ghc.yaml

```diff
@@ -0,0 +1,37 @@
+name: Update GHC
+on:
+  schedule:
+    # run weekly on Thursday
+    - cron:  '0 15 * * THU'
+  workflow_dispatch: # allows manual triggering
+
+jobs:
+  update_ghc:
+    name: GHC ${{ matrix.ghc }} Update
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ghc:
+          - '9.2'
+          - '9.4'
+          - '9.6'
+    steps:
+      - uses: actions/checkout@v3
+      - uses: cachix/install-nix-action@v22
+        with:
+          nix_path: nixpkgs=nixpkgs/default.nix
+      - name: Fetch updates
+        id: ghc_update
+        run: nix shell 'nixpkgs#bazel-buildtools' 'nixpkgs#python3' --command python .github/update-ghc.py ${{ matrix.ghc }}
+      - name: Create Pull Request
+        if: steps.ghc_update.outputs.latest != ''
+        uses: peter-evans/create-pull-request@v5
+        with:
+          commit-message: "[automation] Add GHC version "
+          title: "[update] GHC ${{ steps.ghc_update.latest }}"
+          branch: "automation/update-ghc-${{ matrix.ghc }}"
+          base: master
+          # TODO: needs a token so that workflows for the PR get triggered
+          #token: "${{ secrets.PR_TOKEN }}"
```

Comment:
I think you have the required permissions to set such a token, or do you need help on that?

Reply:
I just wanted to check how other projects have set this up, maybe I could re-use a token -- do we have a bot user maybe?

Otherwise, I'll just create a new PAT and add it as a secret.